pkg/image/image.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2019 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/term"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// SaveToTar takes a list of images and writes them to a tarbal
+func SaveToTar(ctx context.Context, cli *client.Client, images []string, filepath string) error {
+	file, err := os.Create(filepath)
+	if err != nil {
+		return errors.Wrapf(err, "Could not create tarball file '%s'", filepath)
+	}
+	defer file.Close()
+
+	out, err := cli.ImageSave(ctx, images)
+	if err != nil {
+		return errors.Wrap(err, "error saving images to tar")
```

Comment:
I hit this error when trying to run it. I am on version 1.39 of the docker api though locally.

```
./sonobuoy images download
ERRO[0000] couldn't save images to tar: error saving images to tar: Error response from daemon: reference does not exist
```

Reply:
Seems like you don't have all the images downloaded?

Reply:
Worked on linux as well without changes. Hmmm

Reply:
Just resolving this conversation; the thread above about hardcoding the version # has the relevant info. It isnt an OS issue but a docker api version one.